src/User.js

```diff
@@ -1,9 +1,65 @@
+import dayjs from 'dayjs'
+var isBetween = require('dayjs/plugin/isBetween')
+dayjs.extend(isBetween)
+
 class User {
   constructor(jsonObj) {
-    this.id = jsonObj.id
-    this.name = jsonObj.name
-    this.travelerType = jsonObj.travelerType
-    this.trips = []
+    this.id = jsonObj.id;
+    this.name = jsonObj.name;
+    this.travelerType = jsonObj.travelerType;
+    this.trips = [];
+    this.pastTrips = [];
+    this.futureTrips = [];
+    this.currentTrip = null;
+    this.pendingTrips = [];
```

Comment:
I would consider putting these and relevant methods in your Trip class to avoid "God Class"/code smell.